rfc/plugins.md

```diff
@@ -0,0 +1,235 @@
+# Plugins
+
+A plugin is an ember-cli addon that hooks into the ember-cli-deploy
+pipeline in order to add functionality to a deploy. Example addon
+functionality would be uploading assets to S3, writing the index.html to
+Redis, or notifying a Slack channel a deploy has completed.
+
+Plugins should focus on doing a specific task. Ember developers will
+compose and combine multiple plugins to fine tune their deployment
+process.
```

Comment:
Most plugins designed for distribution should focus on a specific task. However, teams wishing to fully customize their deployment process via a private or in-repo addon need not limit the addon to a specific task.